test/ng/compileSpec.js

```diff
@@ -5558,6 +5558,43 @@ describe('$compile', function() {
     });
 
 
+    it('should throw badrestrict on first compilation when restrict is invalid', function() {
+      module(function($compileProvider, $exceptionHandlerProvider) {
+        $compileProvider.directive('invalidRestrictBadString', valueFn({restrict: '"'}));
+        $compileProvider.directive('invalidRestrictFalse', valueFn({restrict: false}));
+        $compileProvider.directive('invalidRestrictTrue', valueFn({restrict: true}));
+        $compileProvider.directive('invalidRestrictObject', valueFn({restrict: {}}));
+        $compileProvider.directive('invalidRestrictZero', valueFn({restrict: 0}));
+
+        // We need to test with the exceptionHandler not rethrowing...
+        $exceptionHandlerProvider.mode('log');
```

Comment:
By default in tests `$exceptionHandler` seems to throw. I had to change that to only log so that we can test that it only invokes `$exceptionHandler` once (on first usage of the directive) and any further use of the directive does not re-log the same error message.